python/nano/src/bigdl/nano/tf/keras/inference/optimizer.py

```diff
@@ -0,0 +1,262 @@
+               might be exposed to the model, which causing data leak.
+        :param validation_data: (optional) An unbatched tf.data.Dataset object for accuracy
+               evaluation. This is only needed when users care about the possible accuracy drop.
+        :param metric: (optional) A tensorflow.keras.metrics.Metric object which is used for
+               calculating accuracy.
+        :param direction: (optional) A string that indicates the higher/lower
+               better for the metric, "min" for the lower the better and "max" for the
+               higher the better. Default value is "max".
+        :param thread_num: (optional) a int represents how many threads(cores) is needed for
+               inference.
+        :param logging: whether to log detailed information of model conversion.
+               Default: False.
+        :param latency_sample_num: (optional) a int represents the number of repetitions
+               to calculate the average latency. The default value is 100.
+        :param includes: (optional) a list of acceleration methods that will be included in the
+               search. Default to None meaning including all available methods. "original" method
+               will be automatically add to includes.
+        :param excludes: (optional) a list of acceleration methods that will be excluded from the
+               search. "original" will be ignored in the excludes.
+        '''
+        # check if model is a nn.Module or inherited from a nn.Module
+        invalidInputError(isinstance(model, Model), "model should be a Keras Model.")
+        invalidInputError(direction in ['min', 'max'],
+                          "Only support direction 'min', 'max'.")
+
+        if not isinstance(model, NanoModel):
+            # turn model into NanoModel to obtain trace and quantize method
+            model = NanoModel(inputs=model.inputs, outputs=model.outputs)
+
+        # get the available methods whose dep is met
+        available_dict: Dict =\
+            available_acceleration_combination(excludes=excludes,
+                                               includes=includes,
+                                               full_methods=self.ALL_INFERENCE_ACCELERATION_METHOD)
+
+        self._direction: str = direction  # save direction as attr
+        # record whether calculate accuracy in optimize by this attr
+        if validation_data is None or metric is None:
+            self._calculate_accuracy = False
+        else:
+            # test whether accuracy calculation works later
+            # make sure dataset don't have batch
+            batched_validation_data = validation_data.batch(batch_size)
+            self._calculate_accuracy = True
+
+        if os.getenv('OMP_NUM_THREADS') is not None:
+            default_threads: int = int(os.getenv('OMP_NUM_THREADS'))  # type: ignore
+        else:
+            # TODO: how to get and control thread num in tf?
```

Comment:
https://www.tensorflow.org/api_docs/python/tf/config/threading

![image](https://user-images.githubusercontent.com/10561966/201448219-15aaa1cd-5cf8-40d6-b4a3-a6161941d3aa.png)

Reply:
I have tried this locally, and get_xxx_threads() return 0 after we set `OMP_NUM_THREADS`.
And we can't set_xxx_threads in optimize(), it will report :
![image](https://user-images.githubusercontent.com/105281011/201663051-198671ae-8e0a-4cb6-a277-be41a9f6bae2.png)